File: Code/Constants/Loadout.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace watchtower.Constants {

    public sealed class Loadout {

        public const short NC_INFILTRATOR = 1;
        public const short NC_LIGHT_ASSAULT = 3;
        public const short NC_MEDIC = 4;
        public const short NC_ENGINEER = 5;
        public const short NC_HEAVY_ASSAULT = 6;
        public const short NC_MAX = 7;

        public const short TR_INFILTRATOR = 8;
        public const short TR_LIGHT_ASSAULT = 10;
        public const short TR_MEDIC = 11;
        public const short TR_ENGINEER = 12;
        public const short TR_HEAVY_ASSAULT = 13;
        public const short TR_MAX = 14;

        public const short VS_INFILTRATOR = 15;
        public const short VS_LIGHT_ASSAULT = 17;
        public const short VS_MEDIC = 18;
        public const short VS_ENGINEER = 19;
        public const short VS_HEAVY_ASSAULT = 20;
        public const short VS_MAX = 21;

        public const short NS_INFILTRATOR = 28;
        public const short NS_LIGHT_ASSAULT = 29;
        public const short NS_MEDIC = 30;
        public const short NS_ENGINEER = 31;
        public const short NS_HEAVY_ASSAULT = 32;
        public const short NS_MAX = 45;

        public static short GetFaction(short loadoutID) {
            if (loadoutID == NC_INFILTRATOR
                    || loadoutID == NC_LIGHT_ASSAULT
                    || loadoutID == NC_MEDIC 
                    || loadoutID == NC_ENGINEER 
                    || loadoutID == NC_HEAVY_ASSAULT
                    || loadoutID == NC_MAX) {
                return Faction.NC;
            } else if (loadoutID == VS_INFILTRATOR
                    || loadoutID == VS_LIGHT_ASSAULT
                    || loadoutID == VS_MEDIC 
                    || loadoutID == VS_ENGINEER 
                    || loadoutID == VS_HEAVY_ASSAULT
                    || loadoutID == VS_MAX) {
                return Faction.VS;
            } else if (loadoutID == TR_INFILTRATOR
                    || loadoutID == TR_LIGHT_ASSAULT
                    || loadoutID == TR_MEDIC 
                    || loadoutID == TR_ENGINEER 
                    || loadoutID == TR_HEAVY_ASSAULT
                    || loadoutID == TR_MAX) {
                return Faction.TR;
            } else if (loadoutID == NS_INFILTRATOR
                    || loadoutID == NS_LIGHT_ASSAULT
                    || loadoutID == NS_MEDIC 
                    || loadoutID == NS_ENGINEER 
                    || loadoutID == NS_HEAVY_ASSAULT
                    || loadoutID == NS_MAX) {
                return Faction.NS;
            }

            return Faction.UNKNOWN;
        }

    }
}

File: src/util/Loadout.ts

export default class LoadoutUtils {

	public static NAME_INFILTRATOR: string = "infiltrator";
	public static NAME_LIGHT_ASSAULT: string = "light assault";
	public static NAME_MEDIC: string = "medic";
	public static NAME_ENGINEER: string = "engineer";
	public static NAME_HEAVY_ASSAULT: string = "heavy assault";
	public static NAME_MAX: string = "MAX";

	public static NC_INFILTRATOR: number = 1;
	public static NC_LIGHT_ASSAULT: number = 3;
	public static NC_MEDIC: number = 4;
	public static NC_ENGINEER: number = 5;
	public static NC_HEAVY_ASSAULT: number = 6;
	public static NC_MAX: number = 7;

	public static TR_INFILTRATOR: number = 8;
	public static TR_LIGHT_ASSAULT: number = 10;
	public static TR_MEDIC: number = 11;
	public static TR_ENGINEER: number = 12;
	public static TR_HEAVY_ASSAULT: number = 13;
	public static TR_MAX: number = 14;

	public static VS_INFILTRATOR: number = 15;
	public static VS_LIGHT_ASSAULT: number = 17;
	public static VS_MEDIC: number = 18;
	public static VS_ENGINEER: number = 19;
	public static VS_HEAVY_ASSAULT: number = 20;
	public static VS_MAX: number = 21;

	public static NS_INFILTRATOR: number = 28;
	public static NS_LIGHT_ASSAULT: number = 29;
	public static NS_MEDIC: number = 30;
	public static NS_ENGINEER: number = 31;
	public static NS_HEAVY_ASSAULT: number = 32;
	public static NS_MAX: number = 45;

	public static isInfiltrator(id: number): boolean {
		return id == this.NC_INFILTRATOR
			|| id == this.TR_INFILTRATOR
			|| id == this.VS_INFILTRATOR
			|| id == this.NS_INFILTRATOR;
	}

	public static isLightAssault(id: number): boolean {
		return id == this.NC_LIGHT_ASSAULT
			|| id == this.TR_LIGHT_ASSAULT
			|| id == this.VS_LIGHT_ASSAULT
			|| id == this.NS_LIGHT_ASSAULT;
	}

	public static isMedic(id: number): boolean {
		return id == this.NC_MEDIC
			|| id == this.TR_MEDIC
			|| id == this.VS_MEDIC
			|| id == this.NS_MEDIC;
	}

	public static isEngineer(id: number): boolean {
		return id == this.NC_ENGINEER
			|| id == this.TR_ENGINEER
			|| id == this.VS_ENGINEER
			|| id == this.NS_ENGINEER;
	}

	public static isHeavy(id: number): boolean {
		return id == this.NC_HEAVY_ASSAULT
			|| id == this.TR_HEAVY_ASSAULT
			|| id == this.VS_HEAVY_ASSAULT
			|| id == this.NS_HEAVY_ASSAULT;
	}

	public static isMax(id: number): boolean {
		return id == this.NC_MAX
			|| id == this.TR_MAX
			|| id == this.VS_MAX
			|| id == this.NS_MAX;
	}

	public static getFactionID(loadoutID: number): number {
        switch (loadoutID) {
            case LoadoutUtils.NC_INFILTRATOR:
            case LoadoutUtils.NC_LIGHT_ASSAULT:
            case LoadoutUtils.NC_MEDIC:
            case LoadoutUtils.NC_ENGINEER:
            case LoadoutUtils.NC_HEAVY_ASSAULT:
            case LoadoutUtils.NC_MAX:
                return 2;

            case LoadoutUtils.TR_INFILTRATOR:
            case LoadoutUtils.TR_LIGHT_ASSAULT:
            case LoadoutUtils.TR_MEDIC:
            case LoadoutUtils.TR_ENGINEER:
            case LoadoutUtils.TR_HEAVY_ASSAULT:
            case LoadoutUtils.TR_MAX:
                return 3;

            case LoadoutUtils.VS_INFILTRATOR:
            case LoadoutUtils.VS_LIGHT_ASSAULT:
            case LoadoutUtils.VS_MEDIC:
            case LoadoutUtils.VS_ENGINEER:
            case LoadoutUtils.VS_HEAVY_ASSAULT:
            case LoadoutUtils.VS_MAX:
                return 1;

            case LoadoutUtils.NS_INFILTRATOR:
            case LoadoutUtils.NS_LIGHT_ASSAULT:
            case LoadoutUtils.NS_MEDIC:
            case LoadoutUtils.NS_ENGINEER:
            case LoadoutUtils.NS_HEAVY_ASSAULT:
            case LoadoutUtils.NS_MAX:
                return 4;

			default: return -1;
        }
    }

	public static getLoadoutName(loadoutID: number): string {
        switch (loadoutID) {
            case LoadoutUtils.VS_INFILTRATOR:
            case LoadoutUtils.NC_INFILTRATOR:
            case LoadoutUtils.TR_INFILTRATOR:
			case LoadoutUtils.NS_INFILTRATOR:
				return this.NAME_INFILTRATOR;

            case LoadoutUtils.NC_ENGINEER:
            case LoadoutUtils.TR_ENGINEER:
            case LoadoutUtils.VS_ENGINEER:
			case LoadoutUtils.NS_ENGINEER:
				return this.NAME_ENGINEER;

            case LoadoutUtils.NC_MAX:
            case LoadoutUtils.TR_MAX:
            case LoadoutUtils.VS_MAX:
			case LoadoutUtils.NS_MAX:
				return this.NAME_MAX;

            case LoadoutUtils.TR_LIGHT_ASSAULT:
            case LoadoutUtils.NC_LIGHT_ASSAULT:
            case LoadoutUtils.VS_LIGHT_ASSAULT:
			case LoadoutUtils.NS_LIGHT_ASSAULT:
				return this.NAME_LIGHT_ASSAULT;

            case LoadoutUtils.TR_MEDIC:
            case LoadoutUtils.NC_MEDIC:
            case LoadoutUtils.VS_MEDIC:
			case LoadoutUtils.NS_MEDIC:
				return this.NAME_MEDIC;

            case LoadoutUtils.TR_HEAVY_ASSAULT:
            case LoadoutUtils.NC_HEAVY_ASSAULT:
            case LoadoutUtils.VS_HEAVY_ASSAULT:
			case LoadoutUtils.NS_HEAVY_ASSAULT:
				return this.NAME_HEAVY_ASSAULT;

			default: return `unchecked loudoutID ${loadoutID}`;
        }

    }

}
